app/jobs/in_person/enrollments_ready_for_status_check/enrollment_pipeline.rb

```diff
@@ -0,0 +1,144 @@
+      end
+
+      # Add information to help with debugging
+      error_extra[:ses_aws_message_id] = ses_message.dig(:mail, :messageId)
+      error_extra[:ses_mail_timestamp] = ses_message.dig(:mail, :timestamp)
+      error_extra[:ses_mail_source] = ses_message.dig(:mail, :source)
+
+      # https://datatracker.ietf.org/doc/html/rfc5322#section-3.6.1
+      error_extra[:ses_rfc_origination_date] = ses_message.
+        dig(:mail, :commonHeaders, :date)&.tap do |date|
+          DateTime.parse(date).to_s
+        end
+      # https://datatracker.ietf.org/doc/html/rfc5322#section-3.6.4
+      error_extra[:ses_rfc_message_id] = ses_message.dig(:mail, :commonHeaders, :messageId)
+
+      # Parse email from content of SES message
+      begin
+        mail = Mail.read_from_string(ses_message[:content])
+        # Depending on how the email is created, we may need to read different
+        # parts of the message
+        if mail.multipart?
+          text_body = mail.text_part&.decoded
+        else
+          text_body = mail.body.decoded
+        end
+      rescue StandardError
+        report_error(err, **error_extra)
+        return false
+      end
+
+      unless text_body.respond_to?(:to_s) && text_body.to_s.present?
+        report_error('Failure occurred when attempting to get email body', **error_extra)
+        return false
+      end
+
+      # Extract enrollment code from email body
+      enrollment_code = email_body_pattern.match(text_body.to_s)&.[](:enrollment_code)
+
+      unless enrollment_code.is_a?(String)
+        report_error(
+          'Failed to extract enrollment code using regex, check email body format and regex',
+          **error_extra,
+        )
+        return false
+      end
+
+      error_extra[:enrollment_code] = enrollment_code
+
+      # Look up existing enrollment
+      id, user_id, ready_for_status_check = InPersonEnrollment.
+        where(enrollment_code:, status: :pending).
+        order(created_at: :desc).
+        limit(1).
+        pick(
+          :id, :user_id, :ready_for_status_check
+        )
+
+      if id.nil?
+        report_error(
+          'Received code for enrollment that does not exist in the database',
+          **error_extra,
+        )
+        return false
+      end
+
+      error_extra[:enrollment_id] = id
+      error_extra[:user_id] = user_id
+
+      # SQS can deliver the message multiple times, so it's expected that
+      # sometimes ready_for_status_check will already be set to true.
+      unless ready_for_status_check
+        # Mark enrollment as ready for the USPS status check.
+        #
+        # Note: There is still a chance of duplicate updates at this point,
+        # but the conditional above should catch most cases.
+        InPersonEnrollment.update(id, ready_for_status_check: true)
+      end
+      return true
+    rescue StandardError => err
+      # Report and re-throw unhandled errors
+      report_error(err, **error_extra)
+      raise err
+    end
```

Comment:
I believe unhandled errors in background jobs already go to NewRelic, do we need to notify directly like this?

Reply:
We do need additional direct error logging here in order to better capture the context of the error; this is most likely to occur after we have captured the enrollment code. However that part could be isolated to CloudWatch if you think that's particularly important.